### PR TITLE
rename fee_token_address into deprecated_fee_token_address

### PR DESCRIPTION
**Pull request by zuphitf**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/830)

### CODE REVIEW

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 6 of 10 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/830)* status: all files reviewed, 4 unresolved discussions (waiting on @zuphitf)

___
*[`crates/blockifier/src/state/state_api.rs` line 45 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-NbnbuD35txUlJHAhmoD:-NbnbuD35txUlJHAhmoE:b-jr1cu3) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/state/state_api.rs#L45)):*
> ```XML
>     // TODO(Dori, 1/7/2023): When a standard representation for large integers is set, change the
>     //    return type to that.
>     fn get_fee_token_balance(
> ```

please add a todo stating that tx version should affect the fee address
___
*[`crates/blockifier/src/transaction/account_transaction.rs` line 448 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-Nbnc2SX2MzRL6MRSto_:-Nbnc2SX2MzRL6MRStoa:b-h9vmix) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/transaction/account_transaction.rs#L448)):*
> ```XML
>         let validate_call_info: Option<CallInfo>;
>         let execute_call_info: Option<CallInfo>;
>         let fee_token_address = block_context.deprecated_fee_token_address;
> ```

intermediate variable really needed here?
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 788 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-NbncCdL1oiDrg4i9ReA:-NbncCdL1oiDrg4i9ReB:b6mzb0v) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/transaction/account_transactions_test.rs#L788)):*
> ```XML
>     state: &mut CachedState<DictStateReader>,
> ) -> TransactionExecutionInfo {
>     let fee_token_address = *block_context.deprecated_fee_token_address.0.key();
> ```

intermediate var needed?
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 811 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-NbncHuE51saWy6dFXBS:-NbncHuE51saWy6dFXBT:b6mzb0v) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/transaction/account_transactions_test.rs#L811)):*
> ```XML
>     #[from(create_state)] state: CachedState<DictStateReader>,
> ) {
>     let fee_token_address = *block_context.deprecated_fee_token_address.0.key();
> ```

intermediate var needed?

**Review by zuphitf — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/830)* status: 9 of 10 files reviewed, 4 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/blockifier/src/state/state_api.rs` line 45 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-NbnbuD35txUlJHAhmoD:-NbnpZGJ5aAbTmikSF5y:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/state/state_api.rs#L45)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

please add a todo stating that tx version should affect the fee address
</blockquote></details>

Done.
___
*[`crates/blockifier/src/transaction/account_transaction.rs` line 448 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-Nbnc2SX2MzRL6MRSto_:-Nbnpm0AC21G6y6wBfXi:baphoo6) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/transaction/account_transaction.rs#L448)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

intermediate variable really needed here?
</blockquote></details>

It's going to be needed once the new tx version is introduced. Figured it's best to add this here already so that the next change will be contained to this line and not below too.
Basically, a similar replacement for all the `let fee_token_address = block_context.deprecated_fee_token_address;` lines.
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 788 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/830#-NbncCdL1oiDrg4i9ReA:-NbnqX9L0aMPlWGaA3DG:b-oaom5f) ([raw file](https://github.com/starkware-libs/blockifier/blob/e38c663b1396a2245a3511da9d0a1f17ab5151a6/crates/blockifier/src/transaction/account_transactions_test.rs#L788)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

intermediate var needed?
</blockquote></details>

depends on whether we're going to want this test to test both v2 and v3 txs or just v2.

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/830#-:-NbnrfQQ2WUIbzt_n0bd:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @zuphitf)
